apps/etl/mutations.py

```diff
@@ -60,54 +61,60 @@
 
-    @strawberry.mutation
-    async def retrigger_transform(
-        self,
-        data: TransformRetriggerInput,
-        info: Info,
-    ) -> MutationResponseType[str]:
-        try:
-            await sync_to_async(run_transform_retrigger)(data)
-            return MutationResponseType(ok=True, result="Successfully retriggered failed transform objects")
-        except Exception:
-            return MutationResponseType(
-                ok=False,
-                errors=_CustomErrorType.generate_message("Fail to retrigger failed transform objects"),
-            )
+    if not failed_transform_objects.exists():
+        raise ValueError("No failed transform objects found for the provided IDs")
+
+    existing_ids = set(Transform.objects.filter(id__in=data.transform_ids).values_list("id", flat=True).distinct())
+
+    missing_ids = set(data.transform_ids) - set(str(id) for id in existing_ids)
+    if missing_ids:
+        raise ValueError(f"IDs not found: {', '.join(missing_ids)}")
+
+    return [obj.id for obj in failed_transform_objects]
+
+
+def validate_retrigger_pipeline(data):
+    failed_extraction_objects = ExtractionData.objects.filter(
+        trace_id__in=data.trace_ids, status=ExtractionData.Status.FAILED
+    )
+    if not failed_extraction_objects.exists():
+        raise ValueError("No failed extraction objects found for the provided trace IDs")
+
+    existing_trace_ids = set(
+        ExtractionData.objects.filter(trace_id__in=data.trace_ids).values_list("trace_id", flat=True).distinct()
+    )
+
+    missing_trace_ids = set(data.trace_ids) - set(str(id) for id in existing_trace_ids)
+    if missing_trace_ids:
+        raise ValueError(f"Trace IDs not found: {', '.join(missing_trace_ids)}")
+
+    return [obj.id for obj in failed_extraction_objects]
 
 
-def run_transform_retrigger(data: TransformRetriggerInput) -> None:
+@shared_task
+def run_transform_retrigger(transform_objects_ids) -> None:
     logger.info("Transform retrigger processing")
-    failed_transform_objects = Transform.objects.filter(id__in=data.transform_ids, status=Transform.Status.FAILED)
 
+    failed_transform_objects = Transform.objects.filter(
+        id__in=transform_objects_ids,
+    )
+    failed_transform_objects.update(status=Transform.Status.PENDING)
     for obj in failed_transform_objects:
         transform_class = source_transform_map[obj.extraction.source]
         transform_class.task.delay(obj.extraction.id)
 
 
-def run_pipeline_retrigger(data: PipelineRetriggerInput) -> None:
+@shared_task
+def run_pipeline_retrigger(extraction_objects_ids) -> None:
     logger.info("Pipeline retrigger processing")
 
     failed_extraction_objects = ExtractionData.objects.filter(
-        trace_id__in=data.trace_ids, status=ExtractionData.Status.FAILED
+        id__in=extraction_objects_ids,
     )
+    failed_extraction_objects.update(status=ExtractionData.Status.PENDING)
```

Comment:
@Rup-Narayan-Rajbanshi check for check for `failed_transform_objects` if it exists